src/main/java/qna/domain/Answer.java

```diff
@@ -111,4 +110,20 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(id);
     }
+
+    public void delete(User loginUser) throws CannotDeleteException {
+        verifyDeletable(loginUser);
+        createHistory(loginUser);
+    }
+
+    private void verifyDeletable(User loginUser) throws CannotDeleteException {
+        if (!isOwner(loginUser)) {
+            throw new CannotDeleteException("다른 사람이 쓴 답변이 있어 삭제할 수 없습니다.");
+        }
+        deleted = true;
+    }
+
+    private DeleteHistory createHistory(User loginUser) {
+        return DeleteHistory.addHistory(ContentType.ANSWER,id,loginUser);
```

Comment:
정적 팩토리 method 활용 👍 

개인적으로 하나 더 의견을 드리자면 
이왕 이렇게 까지 하신거 `addHistory`보다는 더 구체적이면 훨씬 명확할 것 같습니다 😄 
```suggestion
        return DeleteHistory.answer(id,loginUser);
```

이 부분도 참고만 부탁 드립니다 🙇

Reply:
말씀하신것처럼 한 메소드가 answer와 question 두가지의 히스토리를 담당하는것보다 
answer , question으로 분리함으로써 입력인자 갯수도 줄일 수 있고, 메소드명도 더 직관적이어서 좋은것 같습니다. 
도메인 메소드명은 addAnswerDeleteHistoryTo(loginUser)  혹은 addAnswerDeleteHistory(loginUser)

Reply:
보기에 명확하면 좋다고 생각합니다 😄 
명명은 늘 어려운 것 같습니다 😭 
둘 다 괜찮습니다 👍